File: Design Patterns/Behavioral/Strategy/Case 1/Python/RubberDuck.py
from Duck import Duck
from Fly.FlyBehavior import FlyBehavior
from Fly.NoFly import NoFly
from Quak.QuackBehavior import QuackBehavior
from Quak.Squeak import Squeak


class RubberDuck(Duck,FlyBehavior,QuackBehavior):

    def __init__(self) -> None:
        Duck.__init__(self)
        self.setFlyBehavior(NoFly())
        self.setQuakBehavior(Squeak())
    
    def fly(self):
        self.getFlyBehavior().fly()

    def quak(self):
        self.getQuakBehavior().quak()

    def dispay():
        pass

    def swim():
        pass

File: Design Patterns/Behavioral/Strategy/Case 1/PHP/RubberDuck.php
<?php
namespace PHP;
require_once("Quak/QuakBehavior.php");
use PHP\Quak\QuakBehavior;
require_once("Fly/FlyBehavior.php");
use PHP\Fly\FlyBehavior;
require_once("Fly/NoFly.php");
use PHP\Fly\NoFly;
require_once("Quak/Squeak.php");
use PHP\Quak\Squeak;
require_once("Duck.php");
use PHP\Duck;

class RubberDuck extends Duck implements QuakBehavior,FlyBehavior{
    function __construct()
    {
        $this->setFlyBehavior(new NoFly());
        $this->setQuakBehavior(new Squeak());
    }

    function fly()
    {
        $this->getFlyBehavior()->fly();
    }

    function quak()
    {
        $this->getQuakBehavior()->quak();
    }

    function display()
    {
        echo "Displaying RubberDuck";
    }
}